### ISSUE TITLE
Allow use of remote manifests when specifying manifests 

### ISSUE THREAD

**Issue by aaron-prindle** [GoogleContainerTools/skaffold#7558]
### Expected behavior
Skaffold v2 should be able to apply remote manifests (cluster resources) specified in `manifests.rawYaml`

### Actual behavior
Skaffold v2 doesn't currently have the piece of code needed for this ported over, so it doesn't work

**Comment by aaron-prindle** [comment]
This seems to work IIUC:

`skaffold.yaml`
```
aprindle@aprindle-ssd ~/skaffold/examples/remote-getting-started  [fix-7558]$ ls
build.artifacts  Dockerfile  main.go  README.md  skaffold.yaml

aprindle@aprindle-ssd ~/skaffold/examples/remote-getting-started  [fix-7558]$ cat skaffold.yaml 
apiVersion: skaffold/v2beta29
kind: Config
build:
  artifacts:
  - image: skaffold-example
deploy:
  kubectl:
    manifests:
      - https://raw.githubusercontent.com/GoogleContainerTools/skaffold/main/examples/getting-started/k8s-pod.yaml
# deploy:
#   kubectl:
#     manifests:
```

```
aprindle@aprindle-ssd ~/skaffold/examples/remote-getting-started  [fix-7558]$ skaffold-v1.39.1 render -a build.artifacts --default-repo=gcr.io/aprindle-test-cluster 
apiVersion: v1
kind: Pod
metadata:
  name: getting-started
  namespace: default
spec:
  containers:
  - image: gcr.io/aprindle-test-cluster/skaffold-example:v1.38.0-140-g804abffed-dirty@sha256:4d344cb9f87ec41c9a160ff62b685fd2e74968e420099f34ff113ac4d8161adb
    name: getting-started
aprindle@aprindle-ssd ~/skaffold/examples/remote-getting-started  [fix-7558]$ skaffold render -a build.artifacts --default-repo=gcr.io/aprindle-test-cluster 
apiVersion: v1
kind: Pod
metadata:
  name: getting-started
spec:
  containers:
  - image: gcr.io/aprindle-test-cluster/skaffold-example:v1.38.0-140-g804abffed-dirty@sha256:4d344cb9f87ec41c9a160ff62b685fd2e74968e420099f34ff113ac4d8161adb
    name: getting-started
```

**Comment by aaron-prindle** [comment]
I believe this work is actually done atm, there is this `resolveRemoteAndLocal` method in the Generator currently which resolves remote manifests:
https://github.com/GoogleContainerTools/skaffold/blob/main/pkg/skaffold/render/generate/generate.go#L56

Closing this issue for now, tagging @tejal29 in case there is more to be done here as they initially tracked this work in  #7381

**Comment by aaron-prindle** [comment]
IS this referring to the `remoteManifests` field that #7055 also mentions?

**Comment by tejal29** [comment]
yes. There is a filed `RemoteManifest` in `v2alphXX.Deploy.Kubectl` which needs to be ported over to `v3.betaXX.Manifests.Remote`

**Comment by tejal29** [comment]
Also verify, if https://github.com/GoogleContainerTools/skaffold/blob/3fbc7f5012c4abe98bcbe24c3297f48f3cb832cd/pkg/skaffold/deploy/kubectl/kubectl.go#L406 is moved over to render. I think @marlon-gamez already did it

**Comment by tejal29** [comment]
I will make changes for this in #7658

**Comment by aaron-prindle** [comment]
I believe this is fixed now with #7658, closing
